### PR TITLE
Intrepid2: Fix inclusion test

### PR DESCRIPTION
**Pull request by mperego**
@trilinos/intrepid2

## Motivation
A STK test (thanks @alanw0 ) exposed a few issues with the inclusion test.

## Testing

Added tests for `Cell` tools using `float` type instead of `double` that reproduced the errors encountered by the STK test.

### CODE REVIEW

File: packages/intrepid2/src/Shared/Intrepid2_Types.hpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ namespace Intrepid2 {
 
     flt_32 s;
     s.f32 = 1;
-    s.f32++;
+    s.i32++;
     return (s.i32 < 0 ? 1 - s.f32 : s.f32 - 1);
```
Comment (CamelliaDPG) [inline packages/intrepid2/src/Shared/Intrepid2_Types.hpp:60]: That's a good catch!

**Review by CamelliaDPG — APPROVED** [review]
Looks good to me!

**Comment by mperego** [comment]
Thanks @CamelliaDPG !

**Comment by sebrowne** [comment]
This got hit with a bug in Framework unit testing, RETEST is an appropriate fix (that testing has been disabled).

**Comment by mperego** [comment]
@sebrowne it failed in the same way. Do I need to rebase the branch?

**Comment by sebrowne** [comment]
> @sebrowne it failed in the same way.
Don't worry about it, the results will still appear on CDash but I changed that AUE-based Python3 unit test build to not block the merge (we can specify some of the builds as non-blocking, which we used when adding new PR configurations to test them out, and I just set that one back to non-blocking).  The AutoTester will not fail because of that build for now.

**Comment by mperego** [comment]
> > @sebrowne it failed in the same way.
> 
> Don't worry about it, the results will still appear on CDash but I changed that AUE-based Python3 unit test build to not block the merge (we can specify some of the builds as non-blocking, which we used when adding new PR configurations to test them out, and I just set that one back to non-blocking). The AutoTester will not fail because of that build for now.

OK, cool. Thanks.

**Pull request merged**